src/components/modus-wc-logo/modus-wc-logo.tsx

```
@@ -0,0 +1,133 @@
+import { LOGO_VARIANTS, LogoName } from './logo-constants';
+
+/**
+ * A component for displaying Trimble product logos with support for both fixed and scalable sizing.
+ * Provides consistent branding across applications with various product logo options.
+ */
+@Component({
+  tag: 'modus-wc-logo',
+  styleUrl: 'modus-wc-logo.scss',
+  shadow: false,
+  assetsDirs: ['assets'],
+})
+export class ModusWcLogo {
+  private inheritedAttributes: Attributes = {};
+  private themeObserver: MutationObserver | null = null;
+
+  /** Reference to the host element */
+  @Element() el!: HTMLElement;
+
+  /** The name of the logo to display. Accepts values like 'trimble', 'viewpoint_field_view', etc. */
+  @Prop() name!: LogoName;
+
+  /** Show emblem version (icon only) instead of full logo */
+  @Prop() emblem?: boolean = false;
+
+  /** Custom CSS class to apply to the logo container. */
+  @Prop() customClass?: string = '';
+
+  /** The alt text for accessibility. If not provided, defaults to the logo name. */
+  @Prop() alt?: string;
+
+  @State() private isLight: boolean = true;
+
+  componentWillLoad() {
+    this.inheritedAttributes = inheritAriaAttributes(this.el);
+
+    this.isLight = isLightMode();
+
+    // Watch for theme attribute changes (only in browser environment)
+    if (typeof MutationObserver !== 'undefined') {
+      this.themeObserver = new MutationObserver(() => {
+        this.isLight = isLightMode();
+      });
+
+      // Observe the html element for data-theme attribute changes
+      this.themeObserver.observe(document.documentElement, {
+        attributes: true,
+        attributeFilter: ['data-theme'],
+      });
+    }
+  }
+
+  disconnectedCallback() {
+    if (this.themeObserver) {
+      this.themeObserver.disconnect();
+    }
+  }
+
+  private getLogoPath(): string {
+    const logoKey = this.name.toLowerCase().replace(/\s+/g, '_');
+    const logoInfo = LOGO_VARIANTS[logoKey];
+
+    if (!logoInfo) {
+      console.warn(
```

can we remove console warnings

---

Since the logos are loaded from the library and developers may provide invalid logo names or use logos that don't have all theme variants (e.g., missing dark mode), the console warnings serve as helpful debugging tools. Without them:

1. Invalid logo names would fail silently with a blank image
2. Missing theme variants would be hard to diagnose

These warnings only fire during development/debugging when there's an actual configuration issue, not during normal operation. They help developers quickly identify and fix integration problems.